src/components/Sidebar/Sidebar.jsx

```diff
@@ -21,6 +21,12 @@ const LeftDrawerCountStyle = {
 export const Sidebar = ({ logo, children, className, footer, notifications, isFixed = true, onLogoClick }) => {
     const [isLeftDrawerOpen, setIsLeftDrawerOpen] = useState(false);
     const [isRightDrawerOpen, setIsRightDrawerOpen] = useState(false);
+
+    useEffect(() => {
+        setIsLeftDrawerOpen(false); // Close the drawer if notifications changes
+        setIsRightDrawerOpen(false);
+    }, [notifications]);
```

Comment:
The bug is for redirecting to the purchase on click, which I see happening now because the URL changes. So how does this fix that bug?

Reply:
The menu sidebar  renders again and the notifications object which we pass as prop is a new object, we are passing a component.